packages/chat-client/src/types/client.ts

```diff
@@ -79,10 +81,17 @@ export abstract class IChatClient {
   public abstract core: ICore;
   public abstract events: EventEmitter;
   public abstract logger: Logger;
-  public abstract chatInvites: IChatInvites;
-  public abstract chatThreads: IChatThreads;
-  public abstract chatThreadsPending: IChatThreadsPending;
-  public abstract chatMessages: IChatMessages;
+  public abstract chatInvites: IStore<number, ChatClientTypes.Invite>;
```

Comment:
Good call on this 💯 was clearly overkill in the end but wanted to go "by the book" here in terms of types when the road ahead was still pretty unclear :)